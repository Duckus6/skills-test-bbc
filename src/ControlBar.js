import React from "react";

function ControlBar({updateIndex,index, total}) {

    const goBackwards = ()=>{
        updateIndex(Math.max(0,index-1))
    };
    const goForwards = ()=>{
        updateIndex(Math.min(total-1, index+1))
    };
    return(<div>
        <button className={"arrow-button previous"} onClick={goBackwards}>&#8592;</button>
        <span>{index+1} of {total}</span>
        <button className={"arrow-button next"} onClick={goForwards}>&#8594;</button>
    </div>)
}
export default ControlBar;
