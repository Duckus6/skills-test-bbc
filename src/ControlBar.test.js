import React from 'react';
import ControlBar from "./ControlBar";
import {shallow,mount} from "enzyme/";



test("renders",()=>{
    const wrapper = shallow(<ControlBar updateIndex={jest.fn()} index={0} total={5} />);
    expect(wrapper.debug()).toMatchSnapshot();
});

let mockUpdateIndex = null;
describe("Next arrow clicked",()=>{
    beforeEach(()=>{
        mockUpdateIndex = jest.fn();
    });
    test("Index increments", ()=>{
        jest.useFakeTimers();
        const wrapper = mount(<ControlBar updateIndex={mockUpdateIndex} index={0} total={5}/>);
        wrapper.find("button.next").simulate("click");
        wrapper.update();
        console.log(mockUpdateIndex);
        expect(mockUpdateIndex).toHaveBeenCalledWith(1);
    });

    test("Index does not increment when at max index", ()=>{
        jest.useFakeTimers();
        const wrapper = mount(<ControlBar updateIndex={mockUpdateIndex} index={4} total={5}/>);
        wrapper.find("button.next").simulate("click");
        wrapper.update();
        console.log(mockUpdateIndex);
        expect(mockUpdateIndex).toHaveBeenCalledWith(4);
    });
});



describe("Previous arrow clicked",()=>{
    beforeEach(()=>{
        mockUpdateIndex = jest.fn();
    });
    test("Index decrements", ()=>{
        jest.useFakeTimers();
        const wrapper = mount(<ControlBar updateIndex={mockUpdateIndex} index={4} total={5}/>);
        wrapper.find("button.previous").simulate("click");
        wrapper.update();
        console.log(mockUpdateIndex);
        expect(mockUpdateIndex).toHaveBeenCalledWith(3);
    });

    test("Index does not decrement when at max index", ()=>{
        jest.useFakeTimers();
        const wrapper = mount(<ControlBar updateIndex={mockUpdateIndex} index={0} total={5}/>);
        wrapper.find("button.previous").simulate("click");
        wrapper.update();
        console.log(mockUpdateIndex);
        expect(mockUpdateIndex).toHaveBeenCalledWith(0);
    });
});
