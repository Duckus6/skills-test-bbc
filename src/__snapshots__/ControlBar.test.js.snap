// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
"<div>
  <button className=\\"arrow-button previous\\" onClick={[Function: goBackwards]}>
    ←
  </button>
  <span>
    1
  </span>
  <button className=\\"arrow-button next\\" onClick={[Function: goForwards]}>
    →
  </button>
</div>"
`;
